youtube-client/src/components/Page.tsx

```diff
@@ -0,0 +1,141 @@
+import { Button, Grid } from '@mui/material';
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import VideoCard from './Card';
+import SearchInput from './Input';
+import styled from 'styled-components';
+import { useQuery } from 'react-query';
+import youtubeApi from '../constants/api';
+import { Video } from '../interfaces';
+import ReactPaginate from 'react-paginate';
+import { useSwipeable } from "react-swipeable";
+
+const Main = styled.main`
+  max-width: 1200px;
+  margin: 0 auto;
+  padding: 20px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+export default function Page(): JSX.Element {
+  const [value, setValue] = useState('');
+  const [currentValue, setCurrentValue] = useState('');
+  const [offset, setOffset] = useState(4);
+  const [perPage] = useState(4);
+  const [pageCount, setPageCount] = useState(1);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [cards, setCards] = useState([]);
+  const [videos, setVideos] = useState([]);
+  const [nextPageToken, setNextPageToken] = useState('');
+  const [prevPageToken, setPrevPageToken] = useState('');
+  const { isLoading, error, data, refetch } = useQuery<Video[], Error>(
+    'videos',
+    () => handleSubmit(),
+    { enabled: false }
+  );
+
+  useEffect(() => {
+    setPageCount(Math.ceil(videos.length / perPage));
+    setCards(videos.slice(offset - perPage, offset));
+  }, [offset, videos]);
+
+  const handleSubmit = async () => {
+    if (!value) return;
+
+    const response = await youtubeApi.get('/search', {
```

Comment:
From my perspective it would be better to move it to separate file and use as a separate function

Reply:
this way is dynamic and that's cool, it could be also used as getVideos, getViews, etc. which allows you to pass only necessary parameters and use shorter versions with lower possibility to make mistake